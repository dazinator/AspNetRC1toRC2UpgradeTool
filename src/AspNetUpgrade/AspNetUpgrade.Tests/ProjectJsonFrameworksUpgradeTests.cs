using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using AspNetUpgrade.Actions;
using NUnit.Framework;

namespace AspNetUpgrade.Tests
{

    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class ProjectJsonFrameworksUpgradeTests
    {

        public ProjectJsonFrameworksUpgradeTests()
        {

        }

        [TestCase(TestProjectJsonContents.Dnx451Project)]
        [Test]
        public void Can_Apply(string json)
        {
            // arrange
            var testFileUpgradeContext = new TestFileUpgradeContext(json);
            var sut = new AspNetUpgrade.Actions.UpgradeFrameworksJson();

            // act
            sut.Apply(testFileUpgradeContext);
            testFileUpgradeContext.SaveChanges();

            // assert.
            var modifiedContents = testFileUpgradeContext.ModifiedJsonContents;
            Approvals.VerifyJson(modifiedContents);
        }

        [TestCase(TestProjectJsonContents.Dnx451Project)]
        [Test]
        public void Can_Undo(string json)
        {
            // arrange
            var testFileUpgradeContext = new TestFileUpgradeContext(json);
            var sut = new AspNetUpgrade.Actions.UpgradeFrameworksJson();

            // act
            sut.Apply(testFileUpgradeContext);
            sut.Undo(testFileUpgradeContext);

            testFileUpgradeContext.SaveChanges();

            // assert.
            var modifiedContents = testFileUpgradeContext.ModifiedJsonContents;
            Approvals.VerifyJson(modifiedContents);
        }



    }
}