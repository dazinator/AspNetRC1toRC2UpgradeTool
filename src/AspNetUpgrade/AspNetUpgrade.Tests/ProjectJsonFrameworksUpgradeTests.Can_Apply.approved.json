{
    
  "compilationOptions":  {
        
    "emitEntryPoint":  true
  
    },
    
  "dependencies":  {},
    
  "commands":  {
        
    "web":  "Microsoft.AspNet.Server.Kestrel",
        
    "ef":  "EntityFramework.Commands"
  
    },
    
  "frameworks":  {
        
    "net452":  {}
  
    }

}